extensions/config-yaml/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/YamlConfigSourceProvider.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_JAR_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_JAR_PRIORITY;
+
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+public class YamlConfigSourceProvider implements ConfigSourceProvider {
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader) {
+        List<ConfigSource> configSources = new ArrayList<>();
+
+        URL jarMicroprofileConfigYml = classLoader.getResource(MICROPROFILE_CONFIG_YML_FILE);
+        if (jarMicroprofileConfigYml != null) {
+            configSources.add(new YamlConfigSource(jarMicroprofileConfigYml, MICROPROFILE_CONFIG_YML_JAR_PRIORITY));
+        }
+        URL jarApplicationYml = classLoader.getResource(APPLICATION_YML_FILE);
+        if (jarApplicationYml != null) {
+            configSources.add(new YamlConfigSource(jarApplicationYml, APPLICATION_YML_JAR_PRIORITY));
+        }
+
+        URL fsMicroprofileConfigYml = getFileSystemURL(MICROPROFILE_CONFIG_YML_FILE);
```

Comment:
In practice, I don't think people will use the `microprofile-config.yml` file externally, but that's me

Reply:
It could happen for a project relying on a MicroProfile Config implementation and being migrated to Quarkus. And then, having `microprofile-config.yml` in the META-INF directory would definitely help, so I'm moving it as you suggested in your previous comment.